tests/Api/PersistentSpec.hs

```diff
@@ -1,17 +1,115 @@
+{-# LANGUAGE ImpredicativeTypes #-}
+{-# LANGUAGE TypeOperators  #-}
+
 module Api.PersistentSpec where
 
 import Arbitrary
-import Frontend.Html
+import Control.Lens
+import Control.Monad.IO.Class
+import Data.Maybe
+import Data.String.Conversions
+import Servant.Server
+import Test.Hspec
+import Test.QuickCheck
 
-import Control.Applicative ((<$>))
-import Data.Typeable (Typeable, typeOf)
+import Api.Persistent
+import Types
 
-import Test.Hspec -- (Spec, describe, it)
-import Test.QuickCheck -- (Arbitrary(..), Gen, Testable, forAll, property)
 
-import Api.Persistent
+-- | the empty database
+mkEmpty :: IO (Persist :~> IO)
+mkEmpty = mkRunPersist
 
+-- | a database state containing one arbitrary item of each type (idea, user, ...)
+mkInitial :: IO (Persist :~> IO)
+mkInitial = do
+    rp <- mkEmpty
+    generate arbitrary >>= unNat rp . addIdea
+    generate arbitrary >>= unNat rp . addUser
```

Comment:
At some point one might need to make the users first to have the ideas tied to actual users.